app/angular/src/server/angular-cli_utils.js

```diff
@@ -0,0 +1,55 @@
+import fs from 'fs';
+import { basename, dirname, normalize, relative, resolve } from '@angular-devkit/core';
+
+function isDirectory(assetPath) {
+  try {
+    return fs.statSync(assetPath).isDirectory();
+  } catch (e) {
+    return false;
+  }
+}
+
+function getAssetsParts(resolvedAssetPath, assetPath) {
+  if (isDirectory(resolvedAssetPath)) {
+    return {
+      glob: '**/*', // Folders get a recursive star glob.
+      input: assetPath, // Input directory is their original path.
+    };
+  }
+
+  return {
+    glob: basename(assetPath), // Files are their own glob.
+    input: dirname(assetPath), // Input directory is their original dirname.
+  };
+}
+
+export function isBuildAngularInstalled() {
+  try {
+    require.resolve('@angular-devkit/build-angular');
+    return true;
+  } catch (e) {
+    return false;
+  }
+}
+
+export function normalizeAssetPatterns(assetPatterns, dirToSearch, sourceRoot) {
+  if (!assetPatterns || !assetPatterns.length) {
+    return [];
+  }
+
+  return assetPatterns.map(assetPattern => {
+    const assetPath = normalize(assetPattern);
```

Comment:
Hi @igor-dv sorry to nudge you like this. I'm gettinga an error here.
At this point my asset pattern looks like this:
```
{ glob: '**/*',
  input: 'apps/viewer/src/assets',
  output: '/assets' }
```

which throws the following error:
```
TypeError: path.match is not a function
```

Reply:
Thanks, will take a look